enclave-manager/web/src/components/enclaves/CreateEnclave.tsx

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from "react";
+import { KurtosisPackage } from "../../client/packageIndexer/api/kurtosis_package_indexer_pb";
+import { PreloadPackage } from "./PreloadPackage";
+import { ManualCreateEnclaveModal } from "./modals/ManualCreateEnclaveModal";
+import { isDefined } from "../../utils";
+import { ConfigureEnclaveModal } from "./modals/ConfigureEnclaveModal";
+import { KURTOSIS_CREATE_ENCLAVE_URL_ARG } from "../constants";
+import { useLocation, useNavigate } from "react-router-dom";
+
+export const CreateEnclave = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const [configureEnclaveOpen, setConfigureEnclaveOpen] = useState(false);
+  const [kurtosisPackage, setKurtosisPackage] = useState<KurtosisPackage>();
+  const [manualCreateEnclaveOpen, setManualCreateEnclaveOpen] = useState(false);
+
+  useEffect(() => {
+    setManualCreateEnclaveOpen(location.hash === `#${KURTOSIS_CREATE_ENCLAVE_URL_ARG}`);
+  }, [location]);
+
+  const handleManualCreateEnclaveConfirmed = (kurtosisPackage: KurtosisPackage) => {
+    setKurtosisPackage(kurtosisPackage);
+    setManualCreateEnclaveOpen(false);
+    setConfigureEnclaveOpen(true);
+  };
+
+  const handleOnPackageLoaded = useCallback(
+    (kurtosisPackage: KurtosisPackage) => {
+      setKurtosisPackage(kurtosisPackage);
+      setConfigureEnclaveOpen(true);
+    }, []);
+
+  const handleCloseManualCreateEnclave = () => {
+    setManualCreateEnclaveOpen(false);
+    if (isDefined(location.hash)) {
+      navigate(`${location.pathname}${location.search}`);
+    }
+  };
+
+  return (
+    <>
+      <PreloadPackage onPackageLoaded={handleOnPackageLoaded} />
+      <ManualCreateEnclaveModal
+        isOpen={manualCreateEnclaveOpen}
+        onClose={handleCloseManualCreateEnclave}
+        onConfirm={handleManualCreateEnclaveConfirmed}
+      />
```

Comment:
took me a moment to work out how you're getting this open now, but makes sense. In a non SPA I think I'd not be that comfortable driving state changes like this, but for this application where we're relying a lot on react-router-dom to manage state changes this does seem more appropriate.

Reply:
I'm using the url fragment to trigger it to open from any "page". 

Would be (genuinely) interested to hear what you'd propose alternatively and what specifically would not make this a good fit in a not SPA?

Reply:
Cool! This all makes sense. Thanks!